src/compiler/checker.ts

```diff
@@ -3484,6 +3496,39 @@ namespace ts {
                     returnTypeNode = createKeywordTypeNode(SyntaxKind.AnyKeyword);
                 }
                 return createSignatureDeclaration(kind, typeParameters, parameters, returnTypeNode, typeArguments);
+
+                function getDeepSynthesizedCloneAndPaintVisibility<T extends Node>(node: T | undefined): T | undefined {
+                    if (isEntityName(node) || isEntityNameExpression(node)) {
+                        // Paint visible aliases required for copied nodes (so required imports are not elided)
+                        isEntityNameVisible(node, context.enclosingDeclaration);

```

Comment:
Calling `isEntityNameVisible` without using the result is odd. I understand what this is doing, but perhaps its better to split `isEntityNameVisible` into two functions: one that tracks visibility and one that uses the result.

Reply:
That'd just be `hasVisibleDeclarations(resolveNameFromEntityName(entityName, enclosingDeclaration), /*shouldComputeAliasToMakeVisible*/ true)`, I suppose?